### ISSUE TITLE
Add the possibility to disable default `Production` and `Tests` environments

### ISSUE THREAD

**Issue by dmytro-kashcheiev** [SpineEventEngine/base#728]
Usecase: 
Several custom environments. I want to fail fast when none of _MY_ custom environments is suitable for the current runtime. In the current Spine implementation, I will get the default `Production` or `Tests` environment, which I would not like.

- Add the possibility to disable default `Production` and `Tests` environments.
- In addition, it would be great to have a custom exception handler for scenarios when non of the environments match the current runtime. So I can send error notifications via some external system to inform developers about misconfiguration.

**Comment by alexander-yevsyukov** [comment]
@dmytro-kashcheiev, how about registering a `CustomEnvironmentType` (before all the other) that would do the job of the exception handling and then throw an exception, so that the execution would stop?

**Comment by dmytro-kashcheiev** [comment]
@alexander-yevsyukov 

Currently, all `CustomEnvironmentTypes` are registered before the default one. 

For example, in my current case, I use the environment variables to determine if this is production or staging. If the environment variable is "some-random-value" I want Environment API to throw an exception. 

I don't want to cover ALL possible values and have something like a `default` environment. I used to have a `default` environment before. it was `LocalEnvironment` but when I misconfigured production I get `Local` version of the app instead of the `Production` one, this was confusing.

With the current API if I have something like: 
```
.register(MyProduction) // For `some-random-value` this will return `enabled=false`. 
.register(MyStaging) // For `some-random-value` this will return `enabled=false`.
```
The Spine will consider such a situation as `io.spine.base.Production`. 


I can achieve the required fail-fast behavior with the current API. I should do the following: 
```
.register(WrongEnvironment) // And here we should inform developer about missconfiguration. 
.register(MyProduction) // For `some-random-value` this will return `false`. 
.register(MyStaging) // For `some-random-value` this will return `false`.
```
But this looks like a hack.
